external/config-checker/src/main/java/org/wso2/security/tools/utils/Utils.java

```diff
@@ -0,0 +1,601 @@
+
+        if (configurationParser == null) {
+            throw new ConfigCheckerException("unable to find an parser for the given file format "
+                    + fileFormat);
+        }
+        return configurationParser;
+    }
+
+    /**
+     * This method is used to get the list of files in a given product package.
+     *
+     * @param filename
+     * @param productDirectoryPath
+     * @return a list of file paths inside the given directory path with the given file name.
+     */
+    public static List<String> searchConfigFiles(String filename, String productDirectoryPath)
+            throws ConfigCheckerException {
+        List<String> paths = new ArrayList<>();
+        try {
+            Files.walk(Paths.get(productDirectoryPath))
+                    .filter(Files::isRegularFile)
+                    .forEach((f) -> {
+                        String filePath = f.toString();
+                        File file = new File(filePath);
+                        if (file.getName().equals(filename)) {
+                            paths.add(filePath);
+                        }
+                    });
+        } catch (IOException e) {
+            throw new ConfigCheckerException("IO Exception occurred while retrieving the configuration files", e);
+        }
+        return paths;
+    }
+
+    /**
+     * This method is used to get the configuration file name.
+     *
+     * @param propertyNode The file node in the parent.xml or child.xml containing 'id' attribute
+     * @return the file name of the configuration file
+     * @throws ConfigCheckerException
+     */
+    private static String getConfigFileType(Node propertyNode) throws ConfigCheckerException {
+        NamedNodeMap nodeMap;
+        String configFileType = null;
+        if (propertyNode.hasAttributes()) {
+            nodeMap = propertyNode.getAttributes();
+
+            for (int nodeMapIndex = 0; nodeMapIndex < nodeMap.getLength(); nodeMapIndex++) {
+                Node refNode = nodeMap.item(nodeMapIndex);
+
+                if ((refNode.getNodeName().equalsIgnoreCase(FILE_TYPE_TAG))) {
+                    configFileType = refNode.getNodeValue();
+                }
+            }
+            if (configFileType != null) {
+                return configFileType;
+            } else {
+                throw new ConfigCheckerException("type attribute is not defined for <file> element in " +
+                        propertyNode.getParentNode().getNodeName());
+            }
+        } else {
+            throw new ConfigCheckerException("No attributes found for <file> element in " +
+                    propertyNode.getParentNode().getNodeName());
+        }
+    }
+
+    /**
+     * This method is used to get the configuration id from the <config> element.
+     *
+     * @param configNode
+     * @return configuration id
+     * @throws ConfigCheckerException
+     */
+    private static String getConfigID(Node configNode) throws ConfigCheckerException {
```

Comment:
```suggestion
    private static String getConfigId(Node configNode) throws ConfigCheckerException {
```

You'll ave to correct usages of the function.